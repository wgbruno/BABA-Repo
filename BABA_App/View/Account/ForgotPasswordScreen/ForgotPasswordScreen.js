import React, {useState} from 'react';
import {View, Text, StyleSheet, ScrollView} from 'react-native';
import CustomInput from '../../../src/components/CustomInput';
import CustomButton from '../../../src/components/CustomButton';
import { NavigationContainer } from 'react-native';
import { Account } from '../../../Objects/AccountCont';

export default function ForgotPasswordScreen({navigation}){
  const [userName, setUserName] = useState('');

  const emailPassword = () => {
    var account = new Account(userName);
    account.sendPassword();
  };

  const onSignInPress = () => {
    navigation.navigate('SignIn');
  };

  return (
    <ScrollView showsVerticalScrollIndicator={false}>
      <View style={styles.root}>
        <Text style={styles.title}>Email your password</Text>

        <CustomInput
          placeholder="Username"
          value={userName}
          setValue={setUserName}
        />

        <CustomButton text="Send" onPress={emailPassword} />

        <CustomButton
          text="Back to Sign in"
          onPress={onSignInPress}
          type="TERTIARY"
        />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
    padding: 20,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#051C60',
    margin: 10,
  },
  text: {
    color: 'gray',
    marginVertical: 10,
  },
  link: {
    color: '#FDB075',
  },
});
