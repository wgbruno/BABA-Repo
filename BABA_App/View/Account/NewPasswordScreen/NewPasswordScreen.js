import React, {useState} from 'react';
import {View, Text, StyleSheet, ScrollView, Alert} from 'react-native';
import CustomInput from '../../../src/components/CustomInput'
import CustomButton from '../../../src/components/CustomButton';
import { NavigationContainer } from 'react-native';
import { Account } from '../../../Objects/AccountCont';

export default function NewPasswordScreen({navigation}){
  const [userName, setUserName] = useState('');
  const [password, setPassword] = useState('');
  const [newPassword, setNewPassword] = useState('');
  const [rePassword, setRePassword] = useState('');

  const onSubmitPressed = () => {
    const account = new Account(userName, null, password);
    if(newPassword != rePassword){
      Alert.alert("New passwords Do Not Match!", "Please submit passwords again.");
    } else if(account.search()){
      Alert.alert("Username does not exist","The username you entered does not exist. Try again.");
    }else if(account.passCheck()){
      Alert.alert("Incorrect Password","Please reenter your current password.");
    }
    else{
      account.newPassword(newPassword)
      Alert.alert('Password Changed', 'You have succesfully changed your password.');
      navigation.navigate('HomeScreen');
    }
    
  };

  const onSignInPress = () => {
    navigation.navigate('SignIn');
  };

  return (
    <ScrollView showsVerticalScrollIndicator={false}>
      <View style={styles.root}>
        <Text style={styles.title}>Change your password</Text>

        <CustomInput placeholder="Username" value={userName} setValue={setUserName} />
        
        <CustomInput placeholder="Old password" value={password} setValue={setPassword} />

        <CustomInput
          placeholder="Enter your new password"
          value={newPassword}
          setValue={setNewPassword}
        />

        <CustomInput
          placeholder="Reenter your new password"
          value={rePassword}
          setValue={setRePassword}
        />

        <CustomButton text="Submit" onPress={onSubmitPressed} />

        <CustomButton
          text="Back to Sign in"
          onPress={onSignInPress}
          type="TERTIARY"
        />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
    padding: 20,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#051C60',
    margin: 10,
  },
  text: {
    color: 'gray',
    marginVertical: 10,
  },
  link: {
    color: '#FDB075',
  },
});