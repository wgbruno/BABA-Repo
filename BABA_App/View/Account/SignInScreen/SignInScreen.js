import React, {useState} from 'react';
import { View, Text, Image, StyleSheet, useWindowDimensions, ScrollView, Alert } from 'react-native';
import Logo from '../../../assets/images/Logo_1.png';
import CustomInput from '../../../src/components/CustomInput';
import CustomButton from '../../../src/components/CustomButton';
import { NavigationContainer } from 'react-native';
import { Account } from '../../../Objects/AccountCont';

export default function SignInScreen({navigation}){
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const {height} = useWindowDimensions();

  const onSignInPressed = () => {
    var account = new Account(username, password);
    console.log(account.userName);
    if(account.search()){
      Alert.alert("Unable to login","No account matching this Username.");
    } else if(!account.passCheck()){
      Alert.alert("Unable to login","Incorrect Password.");
    } else{
      if(account.logInAccount()){
        Alert.alert("Error","Problem Logging in. Try again.");
      }else{
        if(account.accountType == null){
          Alert.alert("Success", "You have logged in!");
          navigation.navigate('HomeScreen', {paramUserName: account.userName, paramAccountType: account.accountType});
        } else if(account.accountType == 'Admin'){
          Alert.alert("Success", "You have logged in as admin!");
          navigation.navigate('AdminScreen');
        } else if(account.accountType == 'Manager'){
          Alert.alert("Success", "You have logged in as manager!");
          navigation.navigate('ManagerHome', {paramUserName: account.userName});
        }
      }
    }
  };

  const onForgotPasswordPressed = () => {
    navigation.navigate('ForgotPassword');
  };

  const onSignUpPress = () => {
    navigation.navigate('SignUp');
  };

  const onGuest = () => {
    navigation.navigate('HomeScreen', {paramUserName: username});
  };

  
  return (
    <ScrollView showsVerticalScrollIndicator={false}>
      <View style={styles.root}>
        <Image
          source={Logo}
          style={[styles.logo, {height: height * 20}]}
          resizeMode="contain"
        />

        <CustomInput
          placeholder="Username"
          value={username}
          setValue={setUsername}
        />
        <CustomInput
          placeholder="Password"
          value={password}
          setValue={setPassword}
          secureTextEntry
        />

        <CustomButton text="Sign In" onPress={onSignInPressed} />

        <CustomButton text="Guest mode" onPress={onGuest} />

        <CustomButton
          text="Forgot password?"
          onPress={onForgotPasswordPressed}
          type="TERTIARY"
        />

        <CustomButton
          text="Don't have an account? Create one"
          onPress={onSignUpPress}
          type="TERTIARY"
        />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
    padding: 20,
  },
  logo: {
    width: '3000%',
    maxWidth: 400,
    maxHeight: 300,
  },
});
