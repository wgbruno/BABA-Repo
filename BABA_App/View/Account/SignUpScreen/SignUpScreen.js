import React, {useState} from 'react';
import {View, Text, StyleSheet, ScrollView, Alert} from 'react-native';
import CustomInput from '../../../src/components/CustomInput';
import CustomButton from '../../../src/components/CustomButton';
import { NavigationContainer } from 'react-native';
import { Account } from '../../../Objects/AccountCont';


export default function SignUpScreen({navigation}){
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [passwordRepeat, setPasswordRepeat] = useState('');

  const onRegisterPlayer = () => {
    var account = new Account(username, email, password, "N/A", "N/A");
    if(password != passwordRepeat){
      Alert.alert("Passwords Do Not Match!", "Please submit passwords again.")
    } 
    if(account.create()){
      Alert.alert("This username already exists.", "Please try a different username.");
      console.log("Duplicate username");
    } else{
      account.sendEmail();
      navigation.navigate('ConfirmEmail');
    }
  };

  const onSignInPress = () => {
    navigation.navigate('SignIn');
  };

  const onTermsOfUsePressed = () => {
    console.warn('onTermsOfUsePressed');
  };

  const onPrivacyPressed = () => {
    console.warn('onPrivacyPressed');
  };
  

  return (
    <ScrollView showsVerticalScrollIndicator={false}>
      <View style={styles.root}>
        <Text style={styles.title}>Create an account</Text>

        <CustomInput
          placeholder="Username"
          value={username}
          setValue={setUsername}
        />
        <CustomInput placeholder="Email" value={email} setValue={setEmail} />
        <CustomInput
          placeholder="Password"
          value={password}
          setValue={setPassword}
          secureTextEntry
        />
        <CustomInput
          placeholder="Repeat Password"
          value={passwordRepeat}
          setValue={setPasswordRepeat}
          secureTextEntry
        />
        
        <CustomButton text="Register" onPress={onRegisterPlayer} />

        <Text style={styles.text}>
          By registering, you confirm that you accept our{' '}
          <Text style={styles.link} onPress={onTermsOfUsePressed}>
            Terms of Use
          </Text>{' '}
          and{' '}
          <Text style={styles.link} onPress={onPrivacyPressed}>
            Privacy Policy
          </Text>
        </Text>
        <CustomButton
          text="Have an account? Sign in"
          onPress={onSignInPress}
          type="TERTIARY"
        />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
    padding: 50,
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
    color: '#051C60',
    margin: 10,
  },
  text: {
    color: 'gray',
    marginVertical: 30,
  },
  link: {
    color: '#FDB075',
  },
  container: {
    flex:1,
    alignItems: "center",
    justifyContent: "center",
  },
  checkboxContainer: {
    flexDirection: "row",
    marginBottom: 10,
  },
  checkbox: {
    alignSelf: "center",
  },
  label: {
    margin: 8,
  },
});
