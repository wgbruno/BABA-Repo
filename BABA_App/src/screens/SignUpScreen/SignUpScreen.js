import React, {useState} from 'react';
import {View, Text, StyleSheet, ScrollView} from 'react-native';
import CheckBox from '@react-native-community/checkbox';
import CustomInput from '../../components/CustomInput';
import CustomButton from '../../components/CustomButton';

import {useNavigation} from '@react-navigation/core';

const SignUpScreen = () => {
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [passwordRepeat, setPasswordRepeat] = useState('');
  const [isSelected, setSelection] = useState(false);

  const navigation = useNavigation();

  const onRegisterManager = () => {
    navigation.navigate('AddTeam');
  };

  const onRegisterPlayer = () => {
    navigation.navigate('AddPlayer');
  };

  const onSignInPress = () => {
    navigation.navigate('SignIn');

  };

  const onTermsOfUsePressed = () => {
    console.warn('onTermsOfUsePressed');
  };

  const onPrivacyPressed = () => {
    console.warn('onPrivacyPressed');
  };
  

  return (
    <ScrollView showsVerticalScrollIndicator={false}>
      <View style={styles.root}>
        <Text style={styles.title}>Create an account</Text>

        <CustomInput
          placeholder="Username"
          value={username}
          setValue={setUsername}
        />
        <CustomInput placeholder="Email" value={email} setValue={setEmail} />
        <CustomInput
          placeholder="Password"
          value={password}
          setValue={setPassword}
          secureTextEntry
        />
        <CustomInput
          placeholder="Repeat Password"
          value={passwordRepeat}
          setValue={setPasswordRepeat}
          secureTextEntry
        />
        <View style = {styles.container}>
          <View style = {styles.checkboxContainer}>
            <CheckBox
              value = {isSelected}
              onValueChange={setSelection}
              style={styles.checkbox}
            />
            <Text style={styles.label}>I'm a manager.</Text>
          </View>
        </View>
        
        <CustomButton text="Register" onPress={isSelected ? onRegisterPlayer : onRegisterManager} />

        <Text style={styles.text}>
          By registering, you confirm that you accept our{' '}
          <Text style={styles.link} onPress={onTermsOfUsePressed}>
            Terms of Use
          </Text>{' '}
          and{' '}
          <Text style={styles.link} onPress={onPrivacyPressed}>
            Privacy Policy
          </Text>
        </Text>
        <CustomButton
          text="Have an account? Sign in"
          onPress={onSignInPress}
          type="TERTIARY"
        />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
    padding: 50,
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
    color: '#051C60',
    margin: 10,
  },
  text: {
    color: 'gray',
    marginVertical: 30,
  },
  link: {
    color: '#FDB075',
  },
  container: {
    flex:1,
    alignItems: "center",
    justifyContent: "center",
  },
  checkboxContainer: {
    flexDirection: "row",
    marginBottom: 10,
  },
  checkbox: {
    alignSelf: "center",
  },
  label: {
    margin: 8,
  },
});

export default SignUpScreen;
