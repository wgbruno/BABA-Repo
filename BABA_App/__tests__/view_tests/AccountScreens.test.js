import {expect, jest, test} from '@jest/globals';
import renderer from 'react-test-renderer';
import DeleteAccountScreen from '../../View/Account/DeleteAccount/DeleteAccount';
import ForgotPasswordScreen from '../../View/Account/ForgotPasswordScreen/ForgotPasswordScreen';
import NewPasswordScreen from '../../View/Account/NewPasswordScreen/NewPasswordScreen';
import SignInScreen from '../../View/Account/SignInScreen/SignInScreen';
import SignUpScreen from '../../View/Account/SignUpScreen/SignUpScreen';

test("Delete Account Screen", function (){
    const das_tree = renderer.create(<DeleteAccountScreen/>).toJSON();
    expect(das_tree).toMatchSnapshot();
});

test("Forgot Password Screen", function (){
    const fps_tree = renderer.create(<ForgotPasswordScreen/>).toJSON();
    expect(fps_tree).toMatchSnapshot();
});

test("New Password Screen", function (){
    const nps_tree = renderer.create(<NewPasswordScreen/>).toJSON();
    expect(nps_tree).toMatchSnapshot();
});

test("Sign In Screen", function (){
    const sis_tree = renderer.create(<SignInScreen/>).toJSON();
    expect(sis_tree).toMatchSnapshot();
});

test("Sign Up Screen", function (){
    const sus_tree = renderer.create(<SignUpScreen/>).toJSON();
    expect(sus_tree).toMatchSnapshot();
});