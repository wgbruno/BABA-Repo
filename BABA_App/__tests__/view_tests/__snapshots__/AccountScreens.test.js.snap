// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Account Screen 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#051C60",
            "fontSize": 24,
            "fontWeight": "bold",
            "margin": 10,
          }
        }
      >
        Delete Account
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Username"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={Object {}}
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Delete
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Delete All (Testing)
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Forgot Password Screen 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#051C60",
            "fontSize": 24,
            "fontWeight": "bold",
            "margin": 10,
          }
        }
      >
        Email your password
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Username"
          style={Object {}}
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Send
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {},
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              Object {
                "color": "gray",
              },
              Object {},
            ]
          }
        >
          Back to Sign in
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`New Password Screen 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#051C60",
            "fontSize": 24,
            "fontWeight": "bold",
            "margin": 10,
          }
        }
      >
        Change your password
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Username"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Old password"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Enter your new password"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Reenter your new password"
          style={Object {}}
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Submit
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {},
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              Object {
                "color": "gray",
              },
              Object {},
            ]
          }
        >
          Back to Sign in
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Sign In Screen 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 20,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../assets/images/Logo_1.png",
          }
        }
        style={
          Array [
            Object {
              "maxHeight": 300,
              "maxWidth": 400,
              "width": "3000%",
            },
            Object {
              "height": 26680,
            },
          ]
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Username"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={Object {}}
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Sign In
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Guest mode
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {},
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              Object {
                "color": "gray",
              },
              Object {},
            ]
          }
        >
          Forgot password?
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {},
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              Object {
                "color": "gray",
              },
              Object {},
            ]
          }
        >
          Don't have an account? Create one
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Sign Up Screen 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#051C60",
            "fontSize": 28,
            "fontWeight": "bold",
            "margin": 10,
          }
        }
      >
        Create an account
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Username"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Email"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Repeat Password"
          secureTextEntry={true}
          style={Object {}}
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Register
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "gray",
            "marginVertical": 30,
          }
        }
      >
        By registering, you confirm that you accept our
         
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#FDB075",
            }
          }
        >
          Terms of Use
        </Text>
         
        and
         
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#FDB075",
            }
          }
        >
          Privacy Policy
        </Text>
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {},
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              Object {
                "color": "gray",
              },
              Object {},
            ]
          }
        >
          Have an account? Sign in
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
